llama_index/core/llms/types.py

```diff
@@ -108,3 +109,8 @@ class LLMMetadata(BaseModel):
             " locally loaded models, this must be manually specified."
         ),
     )
+    system_role: MessageRole = Field(
```

Comment:
not sure whether this is the best way to switch different system role.
you could also make a dict for system role: {model: system_role_type}
so it can auto select which role based on model

Reply:
I don't think I understand the "dict for system role" suggestion, but it is auto-selecting based on the model's metadata. This line is setting a default so that models that are OpenAI compatible (which makes most of them) don't have to specify this value.

Reply:
I think setting it in the metadata makes sense, especially since it's specific to cohere's api